### PR TITLE
network: introduce Describe() bus method and use it in networkctl

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/network/networkd-json.c [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+        if (sd_device_get_property_value(device, "ID_VENDOR_FROM_DATABASE", &vendor) < 0)
+                (void) sd_device_get_property_value(device, "ID_VENDOR", &vendor);
+
+        if (sd_device_get_property_value(device, "ID_MODEL_FROM_DATABASE", &model) < 0)
+                (void) sd_device_get_property_value(device, "ID_MODEL", &model);
+
+        return json_build(ret, JSON_BUILD_OBJECT(
+                                        JSON_BUILD_PAIR("LinkFile", JSON_BUILD_STRING(link)),
+                                        JSON_BUILD_PAIR("Driver", JSON_BUILD_STRING(link)),
```
Comment (keszybz) [inline src/network/networkd-json.c]: link → driver

File: src/network/networkctl.c [hunk 2]
```diff
@@ -2728,6 +2844,8 @@ static int help(void) {
                "  -s --stats             Show detailed link statics\n"
                "  -l --full              Do not ellipsize output\n"
                "  -n --lines=INTEGER     Number of journal entries to show\n"
+               "     --json=pretty|short|off\n"
+               "                         Generate JSON output\n"
```
Comment (keszybz) [inline src/network/networkctl.c:2851]: `--json=...` affects `networkctl status`, but not `networkctl list`. Do you plan to add that too?

**Review by keszybz — COMMENTED** [review]
Looks great apart from one typo.

**Comment by yuwata** [comment]
@keszybz Thank you for your review.
- "Driver" field was duplicated (and one has typo...). Now, "Driver" field is generated from `Link::driver`, instead of udev `ID_NET_DRIVER` field.
- When `--json=` is set, `list` and `status` commands give same information.
- `list` and `status` commands with `--json=` now supports glob pattern.
- `Interface` field in json format is now sorted by ifindex.

PTAL. (As change is slightly larger than requested, I do not upgrade the green label by myself.)

**Pull request merged**